cmd/boulder-va/main.go

```diff
@@ -85,10 +85,17 @@ func main() {
 	}
 
 	var servers bdns.ServerProvider
-	servers, err = bdns.StartDynamicProvider(c.VA.DNSProvider, 60*time.Second)
+	proto := "udp"
+	if features.Enabled(features.DOH) {
+		proto = "tcp"
+	}
+	servers, err = bdns.StartDynamicProvider(c.VA.DNSProvider, 60*time.Second, proto)
 	cmd.FailOnError(err, "Couldn't start dynamic DNS server resolver")
```

Comment:
```suggestion
	cmd.FailOnError(err, fmt.Sprintf("Couldn't start dynamic %s DNS server resolver", proto))
```

Reply:
I think this isn't needed because the things that could cause `StartDynamicProvider` to fail aren't specific to `proto`. They _may_ be specific to the SRV name we're looking up, but I think that may already show up in the errors? Will have to double check.

Reply:
Here's where the error gets returned listing the whole SRV name, including proto:

https://github.com/letsencrypt/boulder/blob/302e2f0b1510a54cae130c453f1ac866f67f1edb/bdns/servers.go#L273-L277